### PR TITLE
Subscribe to focus and update profile name

### PR DESCRIPTION
**Pull request by ryanlntn**
Updates TalkDetailsScreen to listen for when it's focused and refetch the profile name on return.

update-username

### CODE REVIEW

File: app/models/navigation-store/navigation-store.ts [hunk 1]
```diff
@@ -47,6 +55,7 @@ export const NavigationStoreModel = NavigationEvents.named("NavigationStore")
     dispatch(action: NavigationAction, shouldPush: boolean = true) {
       const previousNavState = shouldPush ? self.state : null
       self.state = RootNavigator.router.getStateForAction(action, previousNavState) || self.state
+      self.fireSubscribers(action, previousNavState, self.state)
```
Comment (ryanlntn) [inline app/models/navigation-store/navigation-store.ts]: For some reason this wasn't hooked up.

File: app/screens/talk-details/talk-details-screen.tsx [hunk 2]
```diff
@@ -194,9 +197,25 @@ export class TalkDetailsScreen extends React.Component<TalkDetailsScreenProps, {
 
   async componentDidMount() {
     const { id } = this.props.navigation.state.params.talk
+    const { addListener, currentRoute } = this.props.navigationStore
     this.fetchComments()
     this.subscribeToComments(id)
     AppState.addEventListener("change", this.handleAppStateChange)
+    this.fetchUserName()
+    this.navListener = addListener("action", () => {
```
Comment (ryanlntn) [inline app/screens/talk-details/talk-details-screen.tsx]: If this is going to be a useful pattern we could add another action like `onFocus` that takes the routeName and a callback to simplify it's use.

**Comment by ryanlntn** [comment]
This fixes #102

File: app/models/navigation-store/navigation-store.ts (hunk 1 above)
Comment (robinheinze) [reply app/models/navigation-store/navigation-store.ts]: Thank you! I tried doing the focus listener and couldn't figure out why it wasn't working.

**Pull request merged**